tests/testsuite/metadata.rs

```diff
@@ -3503,39 +3755,43 @@ fn filter_platform() {
         "features": []
       }
     ],
-    "root": "path+file:[..]foo#0.1.0"
+    "root": "path+[ROOTURL]/foo#0.1.0"
   },
-  "target_directory": "[..]/foo/target",
+  "target_directory": "[ROOT]/foo/target",
   "version": 1,
-  "workspace_root": "[..]/foo",
+  "workspace_root": "[ROOT]/foo",
   "metadata": null
 }
 "#
-            .replace("$ALT_TRIPLE", alt_target)
-            .replace("$HOST_TRIPLE", host_target)
-            .replace("$ALT_DEP", alt_dep)
-            .replace("$CFG_DEP", cfg_dep)
-            .replace("$HOST_DEP", host_dep)
-            .replace("$NORMAL_DEP", normal_dep)
-            .replace("$FOO", &foo),
+                .replace("$ALT_TRIPLE", alt_target)
```

Comment:
In theory, with snapshotting, do we need all of this manual build up?

Reply:
Yeah, I actually tried removing all the manual build parts before, but I thought that might be too aggressive. So, I decided to just carefully maintain the replacements instead.

I'll make the change.

Reply:
Updated!
There's still an ordering issue, though.  Did I miss anything here?

Reply:
I went ahead and reverted the json part, allowing deprecated for now. Let's move forward!